pkg/asset/machines/baremetal/hosts.go

```diff
@@ -135,6 +135,17 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine, userDataS
 			newHost.Spec.PreprovisioningNetworkDataName = networkConfigSecret.Name
 		}
 
+		if host.IsWorker() {
```

Comment:
I think you need this block to be in the `else` for the if statement below (on line 149) for it to work properly. Hosts don't have to be explicitly marked as workers in the install-config.

Reply:
Done